service_client.go

```diff
@@ -68,30 +71,39 @@ func (client *ServiceClient) initReqOpts(url string, JSONBody interface{}, JSONR
 }
 
 // Get calls `Request` with the "GET" HTTP verb.
-func (client *ServiceClient) Get(url string, JSONResponse interface{}, opts *RequestOpts) (*http.Response, error) {
+func (client *ServiceClient) Get(ctx context.Context, url string, JSONResponse interface{}, opts *RequestOpts) (*http.Response, error) {
+	_, span := trace.SpanFromContext(ctx).Tracer().Start(ctx, "GET : "+url)
 	if opts == nil {
 		opts = new(RequestOpts)
 	}
 	client.initReqOpts(url, nil, JSONResponse, opts)
-	return client.Request("GET", url, opts)
+	response, err := client.Request("GET", url, opts)
+	defer span.End()
+	return response, err
 }
 
 // Post calls `Request` with the "POST" HTTP verb.
-func (client *ServiceClient) Post(url string, JSONBody interface{}, JSONResponse interface{}, opts *RequestOpts) (*http.Response, error) {
+func (client *ServiceClient) Post(ctx context.Context, url string, JSONBody interface{}, JSONResponse interface{}, opts *RequestOpts) (*http.Response, error) {
+	_, span := trace.SpanFromContext(ctx).Tracer().Start(ctx, "POST : "+url)
 	if opts == nil {
 		opts = new(RequestOpts)
 	}
 	client.initReqOpts(url, JSONBody, JSONResponse, opts)
-	return client.Request("POST", url, opts)
+	response, err := client.Request("POST", url, opts)
```

Comment:
`ctx` should continue its journey down to the `ServiceProvider` through `client.Request`